### PR TITLE
AGL Housekeeping

### PR DESCRIPTION
**Pull request by alex-latham**
## All Submissions:

* [X] Merge the target branch into the PR branch. Fix any conflicts that might appear.
* [X] Have you checked to ensure that other pull request are approved for merge [Pull Requests](../../../pulls)?
* [X] What stories does the PR include
  - none, just minor fixes throughout the app
### New Feature Submissions:

* [X] Have you added an explanation of what your changes do?
* [X] Have you lint your code locally prior to submission?

### Testing:

* [X] Have you written new tests for your core changes, as applicable?
  - N/A, but tests were updated to reflect changes
* [X] Have you successfully ran tests with your changes locally?
* [X] All new and existing tests passed.

**Test coverage** = 100% models/features
--------------------------------------------------------------------------------
### Change Log
* cart-image is a class now, not an id anymore since cart-images appear throughout a page (update to HTML and CSS files)
* add order-image class to CSS file
* "Logged in as" text is contained within a span associated with new CSS properties in topnav (now it looks better)
* Replaced all flash[:notice] with flash[:error] or flash[:success] throughout app since there is already CSS formatting to make these red and green, respectively
* Replaced several hand-rolled routes with resources. 
  * Also update references to admin_path/merchant_path/profile_path to literal URIs since merchant_path is reserved for get "/merchants/:id", to: "merchants#show".
* Presence of password is only validated when a user is created now (so that we can update a user without specifying a new password on the Edit Profile page)
* Orders index and show pages are now completely formatted for readability
* Remove unnecessary ternary from profile edit page (since only registered users use the form, there is no need to check that there is a @user object)
* Fix count of active items in Items index spec and items model spec
* Add redirect to orders creation_spec to fix failing test (test broke as a result of a US spec, but not in an unexpected way)
* Dates on orders index/show pages are truncated to M/D/YYYY

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -20,9 +20,8 @@ def edit
 
   def update
     @user = current_user
-    @user.update(user_params)
-    if @user.save
-      flash[:notice] = "Your profile has been updated."
```
Comment (Mariana-21) [inline app/controllers/users_controller.rb:25]: Nice refactor

File: app/models/user.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class User < ApplicationRecord
   has_secure_password
-  validates_presence_of :name, :address, :city, :state, :zip, :email, :password
```
Comment (Mariana-21) [inline app/models/user.rb:3]: Nice catch!

File: app/views/profile/orders/index.html.erb [hunk 3]
```diff
@@ -1,11 +1,24 @@
...
-    <%= link_to "Order ##{order.id}", "/profile/orders/#{order.id}" %>
-    <p>Date created: <%= order.created_at %></p>
-    <p>Date updated: <%= order.updated_at %></p>
-    <p>Status: <%= order.status %></p>
-    <p>Item quantity: <%= order.items.length %></p>
-    <p>Grand total: <%= order.grandtotal %></p>
-  </article>
-<% end %>
\ No newline at end of file
+<h1 align = "center">My Orders</h1>
```
Comment (Mariana-21) [inline app/views/profile/orders/index.html.erb:1]: This all looks so clean!

**Review by Mariana-21 — APPROVED** [review]
Real solid clean up!

File: app/views/profile/orders/index.html.erb (hunk 3 above)
Comment (alex-latham) [reply app/views/profile/orders/index.html.erb:1]: I pretty much copy/pasted the existing /orders/:id page lolol. All thanks to Meg for the "template".

File: app/models/user.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
...
+  validates_presence_of :name, :address, :city, :state, :zip, :email
+  validates_presence_of :password, on: :create
```
Comment (tylertomlinson) [inline app/models/user.rb:4]: Thanks for implementing this

File: app/views/users/edit.html.erb [hunk 5]
```diff
@@ -1,23 +1,23 @@
 <h1>Edit Profile</h1>
 
-<%= form_tag ("/profile"), method: :patch do %>
+<%= form_tag "/profile", method: :patch do %>
```
Comment (tylertomlinson) [inline app/views/users/edit.html.erb:3]: Hell yeah. Was planning on doing this today. nice

File: config/routes.rb [hunk 6]
```diff
@@ -3,54 +3,27 @@
...
-
   post "/cart/:item_id", to: "cart#add_item"
   get "/cart", to: "cart#show"
   delete "/cart", to: "cart#empty"
   delete "/cart/:item_id", to: "cart#remove_item"
 
-  get "/orders/new", to: "orders#new"
-  post "/orders", to: "orders#create"
-  get "/orders/:id", to: "orders#show"
+  resources :merchants do
```
Comment (tylertomlinson) [inline config/routes.rb:18]: I will start using resources now

**Review by tylertomlinson — APPROVED** [review]

**Pull request merged**